test/functional/rpc_net.py

```diff
@@ -178,6 +178,19 @@ def test_getpeerinfo(self):
                 "version": 0,
             },
         )
+
+        self.log.info("Check getpeerinfo with nodeid filter")
+        all_peers = self.nodes[0].getpeerinfo()
+        for peer in all_peers:
+            filtered = self.nodes[0].getpeerinfo(peer["id"])
+            assert_equal(len(filtered), 1)
+            assert_equal(filtered[0]["id"], peer["id"])
+            self.log.debug("Filtered peer id {}: OK".format(peer["id"]))
+
+        self.log.info("Check getpeerinfo with nonexistent nodeid filter")
+        nonexistent_id = max(p["id"] for p in all_peers) + 1000
+        assert_equal(self.nodes[0].getpeerinfo(nonexistent_id), [])
```

Comment:
Would it make sense to throw a `JSONRPCError(RPC_MISC_ERROR,"Peer does not exist")` if the peer does not exist, Similar to the [getblockfrompeer](https://github.com/bitcoin/bitcoin/blob/master/src/rpc/blockchain.cpp#L521-L523) instead of sending back an empty array?

where error is thrown in [PeerManagerImpl::FetchBlock](https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L1880)

Reply:
If others think we should return an empty array then sounds good to me!